src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -82,6 +82,64 @@ public T get(final int... keys) {
         return (T)myData[keys[numNestedDimensions]];
     }
 
+    @SuppressWarnings("unchecked")
+    private T leaf(final int key, final Object[] array){
+        return array == null ? null : (T) array[key];
+    }
+
+    private Object[] unroll(final int key, final Object[] array){
+        return array == null ? null : (Object[]) array[key];
+    }
+
+    /**
+     * Specialized version of get for 1 parameter.
+     * Varargs have a large cost because the arg array is allocated every time.
+     * Using a specialized method eliminates that performance problem.
+     */
+    @SuppressWarnings("unchecked")
+    public T get1(final int key0) {
+        return leaf(key0, data);
```

Comment:
Why no bounds checking when you fetch a leaf value? (applies here and below) Seems like you could go out of bounds in `leaf()` just as easily as in `unroll()`, but you don't bounds check the former.